src/cli_driver.rs

```diff
@@ -457,6 +462,22 @@ impl ProcessingSession {
             }
         }
 
+        if let Some(dir) = args.value_of("outdir") {
+
+            output_path = {
+                let tmp = PathBuf::from(dir);
+                match tmp.extension() {
+                    Some(_) => {
+                        tt_warning!(status, "file extension in `-o/--outdir`; only the directory will be used");
+                        tmp.parent().unwrap().to_path_buf()
+                        }
+                    None => tmp,
+                }};
+            if !output_path.exists() { return Err(ErrorKind::Msg(format!("output directory \"{}\" does not exist", output_path.display())).into());}
+        } else {
+            output_path = fs_root.clone().to_path_buf()
```

Comment:
I see, `fs_root` is `Path`. In that case, `fs_root.to_path_buf()` (or `fs_root.to_owned()`) should do.